endpoints/cookie_sync.go

```diff
@@ -192,6 +197,18 @@ func (req *cookieSyncRequest) filterForGDPR(permissions gdpr.Permissions) {
 	}
 }
 
+// filterToLimit will enforce a max limit on cookiesyncs supplied, picking a random subset of syncs to get to the limit if over.
+func (req *cookieSyncRequest) filterToLimit() {
+	if req.Limit <= 0 {
```

Comment:
You can return immediately if the `Limit` is larger than the sync list length. Trims off the more expensive work.

Reply:
I don't think it trims off anything. if `len(req.Bidders)` is under the limit, then we skip the `for` loop and return immediately already. I was thinking of adding such a check, but then thought harder on what the loop was actually doing and came to the conclusion it was unecessary

Reply:
yeah, definitely ignore this now. In the old version it was making an array & copying stuff. Now it's negligible, and if I had to guess would probably be a net negative